### DISCUSSION TITLE
swf to html conversion

### DISCUSSION THREAD

**Discussion by rahimhaji** [airsdk/Adobe-Runtime-Support#963] [Q&A]
Dear friends,

Greetings! i have some eLearning lessons with user interactions. Now its not loading in browser, i need to convert those files in to some format which can load browser (animations with interactions).

Please suggest me to which format i have to convert.

Whether HTML5 is the only format or is there any other format i can convert?

Please suggest me tools to covert as HTML or other format (which will load in browser to play animation with interactions).

Looking for your suggestions and tools.

 

Thanks in advance.

Syed Abdul Rahim

**Comment by JTtheGeek** [comment]
CheerpX is the only solution I am currently aware of

**Comment by ajwfrost** [comment]
[CheerpX](https://leaningtech.com/cheerpx-for-flash/) would allow you to run the content as Flash/SWF based content within a modern browser, and is very useful for scenarios where you don't want to have to deploy software to the end users. We have some other solutions that would require you to distribute a package to the end user that includes the Flash Player just for this use case. But all of these should be seen as temporary, only to be used for a short term (~ 3 years) whilst you migrate to a native web format..

If it was non-interactive content then we can turn SWF into MP4 etc. but for interactive content you'll need to convert it to HTML/CSS/JS perhaps using some framework or tool. It depends a lot on how the content is authored in the first place.. is it AS2 or AS3, is it created in Flash Professional/Animate as a FLA file, does it use Flex, etc.

Worth checking with [IceStone](https://icestonesoft.com/) as well perhaps.

Finally, we (Harman) provide migration services for Flash content as one of our consultancy offerings, so if you wanted to discuss that further, I can put you in touch with the person who coordinates this activity - email me at andrew.frost at harman.com and I'll forward you his details.
